articles/css-in-js-solutions/index.md

```diff
@@ -0,0 +1,143 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/css-in-js-solutions/
+title: CSS-In-JS For Web Components
+description: This article will cover various techniques a developer can use in defining styles within app components during development than maintaining multiple stylesheets for a single app component.
+author: aransiola-ayodele-leom
+date: 2020-11-11T00:00:00-16:00
+topic: []
+excerpt_separator: <!--more-->
+images:
+
+    - url: /engineering-education/css-in-js-solutions/hero.jpg
+      alt: code view on a laptop
+---
+The modern web is now written in components and not pages. CSS was designed to declare styles in different files (.CSS) for each page of your web application, and this structure makes it challenging to maintain multiple stylesheets for multiple pages.
+<!--more-->
+
+### CSS-In-JS For Web Components
+Web components have changed the landscape of development of web applications for front-end developers. The use of web components provides ample opportunities to reuse and build UI components and custom elements that are critical in the development process.
+
+Understanding web components might seem a bit daunting for novice web developers. However, let’s take a web application that has two prime features― a dropdown menu and a navigation menu appears on all pages.
+
+The traditional way of building web applications would require creating multiple CSS, JavaScript and HTML files that have a dropdown and navigation menu as a feature.
+
+This methodology of building web applications makes it difficult to maintain the codebase. But, if the generic features that span the entire web application can be isolated and built-in as components, then each web page will inherit that component and its generic features. This makes the dropdown and navigation menu components reusable, hence Web Components.
+
+### CSS and Web Components
+Styling elements has been the underlying factor for the appeal of frontend web applications, and with it comes a clear understanding of web components and how it relates to CSS.
```

Comment:
**Comment from previous review:** I'd delete this paragraph in favour of starting this section with the next one.